src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx

```diff
@@ -152,6 +155,13 @@ export default function EnvTriggerView({ filteredAppIds, isVirtualEnv }: AppGrou
         }
     }, [filteredAppIds])
 
+    useEffect(() => {
+        if(location.search.includes('approval-node') && filteredWorkflows?.length) {
+            setShowApprovalModal(true)
+            onClickCDMaterial(queryParams.get('approval-node'), DeploymentNodeType.CD, true)
+        }
+    }, [location.search, fliteredWorkflowDataUpdated])
```

Comment:
why are we adding the dependency of fliteredWorkflowDataUpdated

Reply:
As on having the approvalNode in query everytime did not fetch the workflows, so this check has been added , so that when the workflows have been fetched we can fetch the rest data.